cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -91,6 +92,7 @@ public ExecutionMetrics[] retrieveMetrics(ProgramRunId runId) throws MetricRetri
     try {
       return Retries.supplyWithRetries(
         () -> {
+          validateConfiguration();
```

Comment:
Configurations inside `cConf` typically won't change. Why we need to validate on every call to `retrieveMetrics`?

Reply:
This is to validate for HDF config.

Reply:
But we only need to get the values and validate once in the constructor, right?